src/pages/Content/index.js

```diff
@@ -84,247 +42,311 @@ async function getUserInfo() {
+      
+      /*
+      ISSUE 2:
+      The comp variable is made as one of the checks for NOT displaying the toolbar.
+
+      In short,
+        1. If the baseNode and focusNode of the selection are equal then we can display toolbar
+        2. If the baseNode's parentNode is equal to the focusNode's parentNode then we can display the toolbar
+        3. If the focusNode's grandparent (if existing) is the the baseNode's parentNode then we can display the toolbar
+
+        I added #3 to allow links to be highlighted, but it doesn't always work for nested highlights (refer to ISSUE 4)
+
+        For more reference checkout: https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model
+      */
+       let comp =
+        temp.baseNode == temp.focusNode ||
+       temp.baseNode.parentNode == temp.focusNode.parentNode ||
+       temp.baseNode.parentNode == temp.focusNode.parentNode.parentElement;
+       
+       console.log("Can tooltip render for selection:", comp)
+       
+      if (!comp || (selection == lastSelection && isToolTipVisible)) {
+        console.log('I dont want to render at all');
+        e.stopPropagation();
+        return false;
+      } else if (selection.length > 0) {
+        console.log("Rendering the tooltip")
+        //Render the tooltip
+        endX = e.pageX;
+        endY = e.pageY;
+        console.log('start x is ', startY);
+        console.log('end x is ', endY);
+
+        const realStartX = Math.min(startX, endX);
+        const realendX = Math.max(startX, endX);
+
+        const realStartY = Math.min(startY, endY);
+        const realEndY = Math.max(startY, endY);
+        lastSelection = selection;
+        lastSelectionObj = window.getSelection();
+
+        renderToolTip(
+          (realendX - realStartX) / 2 + realStartX,
+          realStartY - (realEndY - realStartY) / 2,
+          selection
+        );
+      } else {
+        closeToolTip();
+      }
+    });
+
+    //Highlight options
+    document.addEventListener('click', (e) => {
+      const parentIdName = e.target.parentNode.getAttribute('id');
+      const currentID = e.target.getAttribute('id');
+      const range = lastSelectionObj != null ? lastSelectionObj.getRangeAt(0) : null;
+      console.log("Range object:", range)
+      const payload = {
+        url: currentURL,
+        userID: currentUserInfo.id,
+        highlightSelection: rangySerializer.serializeRange(range) // Serializes the range into a string to store in DB
+      }
+
+      if (parentIdName == 'highlight' || currentID == 'highlight') {
+
+        /*
+        ISSUE 3: 
+        May need to update the method of update to the DB because someone may just KEEP pushing highlights
+        We need a check before making an API call
+        
+        Might need to check this out: https://github.com/Phildesro123/Reliant#secrets
+        */
+        axios.post('http://localhost:4000/api/websites/addHighlights', payload).then((res) => {
```

Comment:
### ISSUE 3 [minor as of now]:
May need to update the method of update to the DB because someone may just KEEP pushing highlights
We need a check before making an API call
 Might need to check this out: https://github.com/Phildesro123/Reliant#secrets